content-src/selectors/selectors.js

```diff
@@ -36,14 +36,24 @@ module.exports.selectNewTabSites = createSelector(
     state => state.TopSites,
     state => state.FrecentHistory,
     state => state.History,
-    selectSpotlight
+    selectSpotlight,
+    state => state.Blocked
   ],
-  (TopSites, FrecentHistory, History, Spotlight) => {
-    let [topSitesRows, spotlightRows] = dedupe.group([TopSites.rows.slice(0, TOP_SITES_LENGTH), Spotlight.rows]);
+  (TopSites, FrecentHistory, History, Spotlight, Blocked) => {
+
+    // Remove duplicates
+    let [topSitesRows, spotlightRows] = dedupe.group([TopSites.rows.slice(0, TOP_SITES_LENGTH), Spotlight.rows]).map(sites => {
+      return sites.filter(site => !Blocked.urls.has(site.url));
```

Comment:
shouldn't you be removing the blocked sites before deduping so that you end up with the right number of sites in each group? or am I reading this wrong? To me it looks like if one of the TopSites is blocked, it will end up having less than TOP_SITES_LENGTH?